File: 442-find-all-duplicates-in-an-array/find-all-duplicates-in-an-array.java
class Solution {
    public List<Integer> findDuplicates(int[] nums) {
        Set<Integer> unique = new HashSet<>();
        List<Integer> dupli = new ArrayList<>();    
        for(int num:nums){
            if (unique.contains(num)){
                dupli.add(num);
            }
            unique.add(num);

        }
        return dupli;
    }
}

File: 442-find-all-duplicates-in-an-array/find-all-duplicates-in-an-array.go
func findDuplicates(nums []int) []int {
check:=make([]bool, len(nums))
arr:= []int{}
for  i := 0 ;i<len(nums);i++{
    if check[nums[i]-1]{
        arr = append(arr,nums[i])
    }
    check[nums[i]-1]= true

}
return arr
}